### PR TITLE
Fix flaky integration tests by waiting for operations

### PR DESCRIPTION
**Pull request by psteniusubi**
Fixing issues reported in https://github.com/WebOfTrust/signify-ts/issues/201

Implements client side fix for https://github.com/WebOfTrust/keria/issues/147 when checking status of operations that have dependencies

### CODE REVIEW

File: src/keri/app/notifying.ts [hunk 1]
```diff
@@ -61,10 +61,9 @@ export class Notifications {
      * @param {string} said SAID of the notification
      * @returns {Promise<any>} A promise to the result of the deletion
      */
-    async delete(said: string): Promise<any> {
+    async delete(said: string): Promise<void> {
         const path = `/notifications/` + said;
         const method = 'DELETE';
         const res = await this.client.fetch(path, method, null);
```
Comment (lenkan) [inline src/keri/app/notifying.ts:67]:
```suggestion
        await this.client.fetch(path, method, null);
```

File: examples/integration-scripts/utils/test-util.ts [hunk 2]
```diff
@@ -8,23 +8,108 @@ export function sleep(ms: number): Promise<void> {
...
+            }
+        }
+    }
+}
+
+/**
+ * Get status of operation.
+ * If parameter recurse is set then also checks status of dependent operations.
+ */
+async function getOperation<T>(
```
[the remark this answers is not in the document]
Comment (psteniusubi) [reply examples/integration-scripts/utils/test-util.ts:44]: This is currently not exported and only invoked by waitOperation. In waitOperation I wanted to pass through any error from Operations.get, so waitOperation is the one who checks for done.

File: examples/integration-scripts/utils/test-util.ts [hunk 3]
```diff
@@ -8,23 +8,108 @@ export function sleep(ms: number): Promise<void> {
...
+/**
+ * Assert that all notifications were handled
+ */
+export async function assert_notifications(
+    ...clients: SignifyClient[]
+): Promise<void> {
+    for (let client of clients) {
+        let res = await client.notifications().list();
+        for (let note of res.notes.filter((i: any) => i.r === false)) {
+            if (false) {
```
[the remark this answers is not in the document]
Comment (psteniusubi) [reply examples/integration-scripts/utils/test-util.ts:31]: Yes you are correct. 
Currently the integration tests are leaving too many unmarked notifications and I couldn't fix them all. 
I'm not sure what is correct strategy here. I could remove this post-condition check completely.

[the remark this answers is not in the document]
Comment (lenkan) [reply examples/integration-scripts/utils/test-util.ts:31]: Perhaps only add the assertNotifications post condition check for the tests where there are no unmarked notifications? Then create an issue to resolve the remaining ones.

[the remark this answers is not in the document]
Comment (psteniusubi) [reply examples/integration-scripts/utils/test-util.ts]: I added warnNotifications. Then fixed credentials.test.ts so it handles all notifications.

**Comment by lenkan** [comment]
Looks good to me! It would be really nice to get this in. But build is currently complaining about missing unit tests for the added methods in coring.ts.

**Comment by psteniusubi** [comment]
Thanks. I added unit tests. Coverage is looking better now.

File: examples/integration-scripts/credentials.test.ts [hunk 4]
```diff
@@ -299,13 +324,14 @@ test('single signature credentials', async () => {
                 createTimestamp()
             );
 
-        await verifierClient
+        let op = await verifierClient
             .ipex()
             .submitAdmit(verifierAid.name, admit3, sigs3, aend3, [
                 holderAid.prefix,
             ]);
+        waitOperation(verifierClient, op);
```
Comment (lenkan) [inline examples/integration-scripts/credentials.test.ts:332]: Looks like missing `await`. Sorry missed first time.

**Review by lenkan — COMMENTED** [review]
Found a missing await for a `waitOperation` tha tI missed first time around. Last one for me.

**Review by lenkan — APPROVED** [review]

**Comment by psteniusubi** [comment]
Thanks @lenkan. I don't have write access so can you please merge

**Pull request merged**